src/custom-elements/src/components/progress/progress.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, Prop, Watch } from '@stencil/core';
+
+@Component({
+  tag: 'chi-progress',
+  styleUrl: 'progress.scss',
+  scoped: true
+})
+export class Progress {
+  /**
+   *  to set the value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) value: string;
+
+  /**
+   *  to set the min value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) min = '0';
+
+  /**
+   *  to set the max value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) max = '100';
+
+  /**
+   *  to indicate a progress bar state { info, warning, danger, success }.
+   */
+  @Prop({ reflectToAttr: true }) state = 'success';
+
+  @Watch('state')
+  stateValidation(newValue: string) {
+    if (newValue && !['info', 'warning', 'danger', 'success'].includes(newValue)) {
+      throw new Error(`${newValue} is not a valid state for progress. If provided, valid values are: info, warning, danger or success. `);
+    }
+  }
+
+  componentWillLoad() {
+    this.stateValidation(this.state);
+  }
+
+  render() {
+    return (
+      <progress value={this.value ? this.value : undefined}
+        class={this.state ? `-${this.state}` : ''}
+        max={this.max ? this.max : undefined}
+        role="progressbar"
+        aria-valuenow={this.value ? this.value : undefined}
+        aria-valuemin={this.min ? this.min : undefined}
```

Comment:
```suggestion
        aria-valuemin={this.min || undefined}
```

Reply:
@srinath-koneru If we remove min value, aria-valuemin should always be 0